alpha/alpha-server/src/test/java/org/apache/servicecomb/pack/alpha/server/tcc/service/TccTxEventServiceTest.java

```diff
@@ -160,8 +159,8 @@ public void clearUpCompletedTxFromGlobalTxTable() {
 
     tccTxEventService.clearCompletedGlobalTx(1);
 
-    assertThat(participatedEventRepository.findByGlobalTxId(globalTxId).isPresent(), is(false));
-    assertThat(globalTxEventRepository.findByGlobalTxId(globalTxId).isPresent(), is(false));
+    assertThat(participatedEventRepository.findByGlobalTxId(globalTxId).get().isEmpty(), is(true));
```

Comment:
It looks like the JPA API is changed.

Reply:
I think so. 

See the [Null Handling of Repository Methods](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.nullability) section of the Features page. To quote:

> Repository methods returning collections, collection alternatives, wrappers, and streams are guaranteed never to return null but rather the corresponding empty representation